materials/module-1/worksheets/2-peer-to-peer-file-sharing-on-a-mesh.md

```diff
@@ -0,0 +1,80 @@
+Peer-to-peer file sharing on a mesh
+===================================
+
+## Connecting to your Raspberry Pi
+
+1. Verify that the Raspberry Pi has its WiFi adapter(s) proper connected SD card is in place. Power on the device and wait for a solid green LED with a flashing red LED.
+
+1. Connect your computer's WiFi Client to the Raspberry Pi's WiFi Access Point:
+
+        SSID:       <hostname>
+        Password:   password
+
+1. Once connected to the Access Point from your computer, you can connect to it with SSH password `root`:
+
+        $ ssh root@<hostname>.local
+
+    For example, on `bloor` the command would be `ssh root@bloor.local`. You can optionally connect to the IP address directly without using mDNS `ssh root@10.0.0.1`.
+
+1. Find a Raspberry Pi near you can try to ping it to verify the mesh network interface is working. For example:
+
+        root@bloor:~# ping college.local
+
+    Hit `Ctrl + C` to stop the ping.
+
+You are now a mesh node operator running commands from your pet Raspberry Pi, which is connected to the rest of the mesh network we just formed in this room! For the rest of this activity, we will continue to use `bloor` to represent your node, and `college` to represent another node near you. Be sure to replace these with the name of your own pet Raspberry Pi. 
+
+## Running the InterPlanetary File System (IPFS)
+
+1. Start IPFS in the background:
+
+        root@bloor:~# docker load --input ~/docker/tomeshnet-ipfs-0.1.tar
+        root@bloor:~# docker run --name ipfs --network host --detach tomeshnet/ipfs:0.1 
+
+1. Initiate an interactive session and write a message to the peer-to-peer filesystem:
+
+        root@bloor:~# docker exec -it ipfs sh
+        / # echo "Hello World" | ipfs add
+
+1. Using the content hash from the `ipfs add` output, which is uniquely derived from your `Hello World` text, read the content back:
+
+        / # ipfs cat QmWATWQ7fVPP2EFGu71UkfnqhYXDYH566qy47CnJDgvs8u
+
+1. We are also running an ipfs-to-http gateway that can be accessed from clients connected to the Access Point of the Raspberry Pi. You can use a URL formatted like this to access ipfs content from the client browser:
```

Comment:
Maybe frame this as "Next, go you your browser and access the ipfs content you just added! 

Each node is also running an ...